esmvaltool/preprocessor/_time.py

```diff
@@ -61,6 +62,14 @@ def extract_time(cube, start_year, start_month, start_day, end_year, end_month,
             time=lambda t: (t_1 < time_units.date2num(t.point) < t_2))
 
     cube_slice = cube.extract(constraint)
+    if cube_slice is None:
+        start_cube = str(cube.coord('time').points[0])
+        end_cube = str(cube.coord('time').points[-1])
+        raise ValueError("Time slice {} to {} is outside \
```

Comment:
Don't use backslashes, strings are automatically concatenated when they are between brackets, e.g. `raise ValueError("a" "b")` will work just fine.

Reply:
In fact, now that we're dropping Python 2, you could even use an [f-string](https://docs.python.org/3/whatsnew/3.6.html#pep-498-formatted-string-literals):
```
raise ValueError(
    f"Time slice {start_date} to {end_date} is outside cube time bounds "
    f"{start_cube} to {end_cube}.")
```

Reply:
getting a syntax error with the f-string (lovely name btw :grin: ) but guessing my propector is a bit old

Reply:
actually CodaCI doesn't like it either

Reply:
ow wait that coz py2 integration is still on for this pr, nevermind